### PR TITLE
bows and arrows, or How I Learnt To Stop Worrying And Love The Lodge

### PR DESCRIPTION
**Pull request by lolman360**
adds bows, which use the charge mechanic from eguns

charge mechanic can now be used for non-energy guns

support for supereffective damage on projectiles, allowing true 'hunting' weapons without overstacking AP

new mods for bows (just 2 atm, fast charge less damage and slow charge more damage

wip, still need sprites

## Changelog
:cl: lolman360, Churro
add: bows, bow mods
refactor: minor projectiles/ charge firemode tweaks
/:cl:

### CODE REVIEW

**Comment by ThePainkiller** [comment]
Would be nice if bow/crossbow projectiles had a different sound than the throwing one though...

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -125,6 +125,8 @@
 		hit_impact(P.get_structure_damage(), hit_dir)
 		for(var/damage_type in P.damage_types)
 			var/damage = P.damage_types[damage_type]
+			if(faction in P.supereffective_factions)
+				damage *= P.supereffective_mult
```
Comment (hex37) [inline code/modules/mob/living/living_defense.dm:129]: Hey, is there any way you could get this supereffective mult outside of our bullet_act proc? An O(n) search loop here will probably slow things down a little.

Comment (lolman360) [reply code/modules/mob/living/living_defense.dm:129]: I can definitely do something like
```
if(LAZYLEN(P.supereffective_factions))
    loop
```

**Comment by cdb-is-not-good** [comment]
Cool! Serious balance concerns with this, however. If I am reading right, these numbers are very high - and have super_effective bonuses against neutral, which...is every player,.

**Comment by ThePainkiller** [comment]
Could we get the very base _"simple bow"_ parent on a crafting recipe available for anyone on the guns tab?

Would also be good if they could be worn on the suit slot, instead of having to take only the backpack slot, as you can still wrap bows around your chest.

**Comment by lolman360** [comment]
> Cool! Serious balance concerns with this, however. If I am reading right, these numbers are very high - and have super_effective bonuses against neutral, which...is every player,.
balance is based around bonerifle which has similar crafting costs (20 bone 10 plasteel vs 10 bone + nightmare stalker special loot)- it's around as wieldly

assumed players are a separate faction, fixed in next update, it's now typepath based
supereffectivity re: factions is being moved to the `item` var `effective_factions` and `damage_mult` since projectiles are items

**Comment by lolman360** [comment]
> Could we get the very base _"simple bow"_ parent on a crafting recipe available for anyone on the guns tab?

done

> Would also be good if they could be worn on the suit slot, instead of having to take only the backpack slot, as you can still wrap bows around your chest.

they can be worn in suit slot

**Comment by cdb-is-not-good** [comment]
> balance is based around bonerifle which has similar crafting costs (20 bone 10 plasteel vs 10 bone + nightmare stalker special loot)- it's around as wieldly

I'll have to give it a test to see. The numbers feel fairly high and it's worth noting that if the bows are balanced around the baroque(a hunter exclusive) without being hunter exclusive, then it gives a once powerful tool of the hunters to many. But I'll playtest it tonight when I get home just to see

**Comment by cdb-is-not-good** [comment]
Playtested a bit. Inhand for bows needs work. Should change the firing decal so that it's not the same as the guns(looks weird.).
Bows should be two handed, not one handed - for the sake of sanity. Holding bow in one hand, loading arrow, and drawing string back is...ehhhh? I stand by what i said about serrated arrows and their damage values, this is DEFINITELY not "as wieldy as the Baroque" - which requires open bolt, load shot, two hand, fire, unhand, open bolt - etc.

**Comment by lolman360** [comment]
>The numbers feel fairly high and it's worth noting that if the bows are balanced around the baroque(a hunter exclusive) without being hunter exclusive, then it gives a once powerful tool of the hunters to many. 

hunting arrows and the recurve bow are lodge exclusive crafts


> Playtested a bit. Inhand for bows needs work. Should change the firing decal so that it's not the same as the guns(looks weird.)

easy enough

>Bows should be two handed, not one handed - for the sake of sanity. Holding bow in one hand, loading arrow, and drawing string back is...ehhhh? I stand by what i said about serrated arrows and their damage values, this is DEFINITELY not "as wieldy as the Baroque" - which requires open bolt, load shot, two hand, fire, unhand, open bolt - etc.

they should be twohanded, i might've changed it back for testing. i personally found drawing back to be a little more restrictive and absolutely more time-consuming than opening/closing bolt

you haven't actually said anything about serrated arrows yet, but 65 damage 0 ap might be a little much. all values are still up in the air at this point, but for reference serrated arrows are balanced ~around slugs where all the ap was traded for damage. they have a comparable range (5 vs 6)

there's the thing about ammo being renewable, which I probably haven't taken into account re:powerbudgeting

**Comment by cdb-is-not-good** [comment]
Sounds good, we'll iron  out the more serious nitty-gritty of balance once it's actually in play and we can see how they get used. Once the conflicts are resolved lmk and we'll merge this - barring any further points of balance.

**Comment by lolman360** [comment]
Won't be good to merge until I get a sprite for the Guild compound bow, but I'll resolve the conflicts soon. Sprites in general are a little rough IMO, but I can't do any better.

**Comment by Dongels** [comment]
I'm like 90% sure Nayu had some bow sprites..or at some point we had someone post some good looking ones on discord...might of been Gen now that I think about it...hmm.   *shrug*

**Comment by ThePainkiller** [comment]
Suggestion: Back slot icon states

**Comment by lolman360** [comment]
waiting on sprites

**Comment by ThePainkiller** [comment]
Why is the back icon a crossbow though?

image

**Comment by lolman360** [comment]
> Why is the back icon a crossbow though?
> 
> image

used wrong file when resolving conflicts

**Pull request merged**